src/main/java/core/basesyntax/IsoscelesTrapezoidSupplier.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class IsoscelesTrapezoidSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomIsoscelesTrapezoid() {
+        return new IsoscelesTrapezoid(size.getRandomSize(), size.getRandomSize(),
+                size.getRandomSize(), color.getRandomColor());

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. Instead of passing `size.getRandomSize()` directly, assign it to a variable with a meaningful name and then pass that variable to the constructor.